src/quacc/settings.py

```diff
@@ -552,3 +552,33 @@ def change_settings(changes: dict[str, Any]) -> QuaccSettings:  # type: ignore
     finally:
         for attr, original_value in original_values.items():
             setattr(SETTINGS, attr, original_value)
+
+
+def change_settings_wf(
+    func: Callable, changes: dict[str, Any], decorator: Callable
+) -> Callable:
+    """
+    Temporarily change an attribute of an unwrapped object, and return the redecorated object.
+    This is suitable when using workflow management.
+
+    Parameters
+    ----------
+    func
+        The function to update
+    changes
+        Dictionary of changes to make formatted as attribute: value.
+    decorator
+        The decorator associated with `func`.
+
+    Returns
+    -------
+    Callable
+        The updated function.
+    """
+    from quacc import change_settings, strip_decorator
+
+    def concatenated(*args, **kwargs):
```

Comment:
Can you see if decorating `concatenated` with `@wraps` still causes the tests to pass? It is important to retain the original function's metadata: https://docs.python.org/3/library/functools.html#functools.wraps

Reply:
Yes, it passes all the tests under `parsl` in my local. I added a new commit.